### PR TITLE
Add MQTT lightweight files with CONNECT implementation

### PR DESCRIPTION
**Pull request by gordonwang0**
Implements function for generating MQTT connect packets.

Todo: Move common type definitions (network transport, buffer, etc.) to common header.

### CODE REVIEW

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 1]
```diff
@@ -1,9 +1,14 @@
+#ifndef MQTT_LIGHTWEIGHT_H
+#define MQTT_LIGHTWEIGHT_H
+
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 
 #include "config.h"
 
+#define MQTT_PACKET_TYPE_CONNECT         ( ( uint8_t ) 0x10U )
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:10]: Is this needed in header file or can be moved to implementation file?

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+
 struct MQTTFixedBuffer;
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:12]: Where are these defined? Are these opaque to the user of this header?

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 3]
```diff
@@ -1,16 +1,259 @@
...
+        remainingLength = remainingLength / 128U;
+
+        /* Set the high bit of this byte, indicating that there's more data. */
+        if( remainingLength > 0U )
+        {
+            UINT8_SET_BIT( lengthByte, 7 );
+        }
+
+        /* Output a single encoded byte. */
+        *pLengthEnd = lengthByte;
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:73]: Does the caller ensure that the memory is not overflown?

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 4]
```diff
@@ -1,16 +1,259 @@
...
+    connectPacketSize += pConnectInfo->clientIdentifierLength + sizeof( uint16_t );
+
+    /* Add the lengths of the will message and topic name if provided. */
+    if( pWillInfo != NULL )
+    {
+        connectPacketSize += pWillInfo->topicNameLength + sizeof( uint16_t ) +
+                             pWillInfo->payloadLength + sizeof( uint16_t );
+    }
+
+    /* Add the lengths of the user naem and password if provided. */
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:125]: naem --> name

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 5]
```diff
@@ -1,16 +1,259 @@
...
+                                        const MQTTPublishInfo_t * const pWillInfo,
                                         size_t * const pRemainingLength,
                                         size_t * const pPacketSize )
 {
-    return MQTTSuccess;
+    MQTTStatus_t status = MQTTSuccess;
+    size_t remainingLength;
+
+    /* The CONNECT packet will always include a 10-byte variable header. */
+    size_t connectPacketSize = 10U;
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:113]: Is it better to have a #define for 10, something like ```#define CONNECT_PACKET_MIN_LEN 10```?

File: libraries/standard/mqtt/src/mqtt_lightweight.c (hunk 4 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/mqtt_lightweight.c:125]: fixed

File: libraries/standard/mqtt/src/mqtt_lightweight.c (hunk 3 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/src/mqtt_lightweight.c:73]: Yes

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 2 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:12]: They're not opaque.  They're defined in this file.

Some of the structs in this file have members that are other structs.  This requires forward declarations of those structs.  To be consistent, I forward declared everything.

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 1 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:10]: It's needed here.  The application may match `type == MQTT_PACKET_TYPE_CONNECT`.

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 6]
```diff
@@ -1,9 +1,14 @@
+#ifndef MQTT_LIGHTWEIGHT_H
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:22]: Added

**Pull request merged**